### PR TITLE
`youtu.be` URL is not embedded correctly

### PR DESCRIPTION
**Pull request by simong**
When adding a `youtu.be` URL (e.g. https://youtu.be/FYWLiGOBy1k), the link does not embed correctly in the content profile page:

screen shot 2015-03-12 at 10 46 43 am

However, when using the full youtube URL (https://www.youtube.com/watch?v=FYWLiGOBy1k&feature=youtu.be), embedding is working as expected.

As the `youtu.be` and `youtube.com` URLs are essentially the same, they should both embed as expected.

### CODE REVIEW

**Comment by simong** [comment]
Attached a PR that does two things:
 * A short URL expander endpoint
 * Support for the new Youtube APIs as the old ones have been turned off (since the 20th of April)

That last one does requires us to configure a server key so we can talk to the new APIs.

I'll be submitting a PR to jquery.oembed and 3akai-ux shortly

File: node_modules/oae-preview-processor/tests/test-previews.js [hunk 1]
```diff
@@ -909,6 +909,71 @@ describe('Preview processor', function() {
...
+                            }
+                        }
+                    ]
+                });
+        };
+
+        /**
          * Test that verifies the youtube processor and assures that metadata is retrieved/set.
          */
         it('verify youtube link processing works', function(callback) {
```
Comment (mrvisser) [inline node_modules/oae-preview-processor/tests/test-previews.js]: @simong do you know why this test wasn't able to catch the fact that the API was turned off?

Comment (simong) [inline node_modules/oae-preview-processor/tests/test-previews.js]: Although the official end-of-life date was a while ago, the API has only recently defaulted all its responses.

From a travis build:
```
1) Preview processor Preview generation verify youtube link processing works:
      Uncaught AssertionError: "https://youtube.com/devicesupport" == "How to prounounce \"Apereo\""
      + expected - actual
      +"How to prounounce \"Apereo\""
      -"https://youtube.com/devicesupport"
      
      at /home/travis/build/oaeproject/Hilary/node_modules/oae-preview-processor/tests/test-previews.js:925:24
      at /home/travis/build/oaeproject/Hilary/node_modules/oae-preview-processor/tests/test-previews.js:262:29
      at Request._callback (/home/travis/build/oaeproject/Hilary/node_modules/oae-rest/lib/util.js:9:8822)
      at Request.self.callback (/home/travis/build/oaeproject/Hilary/node_modules/oae-rest/node_modules/request/request.js:121:22)
      at Request.emit (events.js:98:17)
      at Request.<anonymous> (/home/travis/build/oaeproject/Hilary/node_modules/oae-rest/node_modules/request/request.js:978:14)
      at Request.emit (events.js:117:20)
      at IncomingMessage.<anonymous> (/home/travis/build/oaeproject/Hilary/node_modules/oae-rest/node_modules/request/request.js:929:12)
      at IncomingMessage.emit (events.js:117:20)
      at _stream_readable.js:938:16
```

**Comment by mrvisser** [comment]
@simong I'm getting a better thumbnail with these changes, but isn't there supposed to be something that points oEmbed to the URL that this PR is introducing for the long url expansion?

**Comment by simong** [comment]
Yes, I still need to submit a UI pr for that

**Comment by mrvisser** [comment]
There is at least one build failure relating to this:

```
1) Long url verify it expands short URLs:
     TypeError: Object #<Object> has no method 'expandUrl'
      at Context.<anonymous> (/home/travis/build/oaeproject/Hilary/node_modules/oae-preview-processor/tests/test-longurl.js:71:26)
      at Test.Runnable.run (/home/travis/build/oaeproject/Hilary/node_modules/mocha/lib/runnable.js:233:15)
      at Runner.runTest (/home/travis/build/oaeproject/Hilary/node_modules/mocha/lib/runner.js:390:10)
      at /home/travis/build/oaeproject/Hilary/node_modules/mocha/lib/runner.js:473:12
      at next (/home/travis/build/oaeproject/Hilary/node_modules/mocha/lib/runner.js:315:14)
      at /home/travis/build/oaeproject/Hilary/node_modules/mocha/lib/runner.js:325:7
      at next (/home/travis/build/oaeproject/Hilary/node_modules/mocha/lib/runner.js:260:23)
      at Object._onImmediate (/home/travis/build/oaeproject/Hilary/node_modules/mocha/lib/runner.js:292:5)
      at processImmediate [as _immediateCallback] (timers.js:345:15)
```

I think I'll hold off further review until the UI counterpart has a PR.

**Comment by simong** [comment]
The test failure is because https://github.com/oaeproject/oae-rest/pull/17 needs to be merged in. I'm not quite sure why the other tests are failing. I don't think it's related

**Comment by simong** [comment]
Assigning back to @mrvisser for review. Note that this can be merged independently from the UI PR

**Comment by mrvisser** [comment]
Can you please rebase this with master and re-enable d0973a0 ?

File: node_modules/oae-preview-processor/tests/test-previews.js [hunk 2]
```diff
@@ -910,15 +910,82 @@ describe('Preview processor', function() {
...
+                                'liveBroadcastContent': 'none',
+                                'localized': {
+                                    'title': 'How to prounounce \"Apereo\"',
+                                    'description': 'Here is Ian Dolphin, the Executive Director of the Apereo Foundation, with the official pronunciation of the word \"Apereo\".'
+                                }
+                            }
+                        }
+                    ]
+                });
+        };
```
Comment (mrvisser) [inline node_modules/oae-preview-processor/tests/test-previews.js]: I'm kind of on the fence about switching to mock this endpoint, as I think it means we can't catch another API breakage until users report it in production. Why is it necessary?

**Comment by mrvisser** [comment]
Back to @simong for feedback

File: node_modules/oae-preview-processor/tests/test-previews.js (hunk 2 above)
Comment (simong) [inline node_modules/oae-preview-processor/tests/test-previews.js]: That's what we used to do originally. But APIs like slideshare or Flickr were intermittently failing the builds due to network issues or just being down in general (https://github.com/oaeproject/Hilary/pull/915)

**Pull request merged**